ocaml/testsuite/tests/typing-unique/unique.ml

```diff
@@ -212,9 +212,9 @@ val x : string = "foo"
 (* Top-level must be many *)
 let once_ foo = "foo"
 [%%expect{|
-Line 1, characters 4-21:
+Line 1, characters 16-21:
 1 | let once_ foo = "foo"
-        ^^^^^^^^^^^^^^^^^
+                    ^^^^^
```

Comment:
This error message was a bit clearer when the ghost location spanned from the start of `once_` to the end of `"foo"`. Can we make `ghexp_with_modes` continue to take a location argument? Then, `ghexp_with_modes` can call `ghostify` on that location before passing it to `mkexp_with_mode` &mdash; if we do this, I'm pretty sure that we can avoid having the `?loc` argument be optional in `mkexp_with_mode` and the functions it calls.

Reply:
The error message is restored now. Also I found that the nameing convention is either `mkexp*` where `make_loc` is called, or `ghexp*` where `ghost_loc` is called - there is no binding where the callsite passes a location that already knows ghost-ness. So to make everything more consistent, I also changed `mkexp_with_mode` to `ghexp_with_mode` that always calls `ghost_loc` - I don't think it makes much difference, and probably less confusing anyway.